### PR TITLE
refactor: adds support for auto-generated round ids

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR refactors AbstractRound such that round_id is no longer required to be set by the developer on the child class. The developer can still do so, as backwards compatibility is maintained in that regard.

## Fixes

-

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/base.py [hunk 1]
```diff
@@ -906,6 +908,26 @@ def _check_class_attributes(self) -> None:
         if not hasattr(self, "synchronized_data_class"):
             logging.warning(f"No `synchronized_data_class` set on {self}")
 
+    @classmethod
```
Comment (DavidMinarsch) [inline packages/valory/skills/abstract_round_abci/base.py:911]: symmetric to behaviour side

File: packages/valory/skills/safe_deployment_abci/rounds.py [hunk 2]
```diff
@@ -70,7 +70,6 @@ def safe_contract_address(self) -> str:
 class RandomnessSafeRound(CollectSameUntilThresholdRound):
     """A round for generating randomness"""
 
-    round_id = "randomness_safe"
```
Comment (DavidMinarsch) [inline packages/valory/skills/safe_deployment_abci/rounds.py:73]: the benefit

File: packages/valory/skills/safe_deployment_abci/tests/test_behaviours.py [hunk 3]
```diff
@@ -303,7 +303,7 @@ class TestValidateSafeBehaviour(BaseValidateBehaviourTest, SafeDeploymentAbciBas
     """Test ValidateSafeBehaviour."""
 
     behaviour_class = ValidateSafeBehaviour
-    next_behaviour_class = make_degenerate_behaviour(FinishedSafeRound.round_id)
+    next_behaviour_class = make_degenerate_behaviour(FinishedSafeRound.auto_round_id())
```
Comment (DavidMinarsch) [inline packages/valory/skills/safe_deployment_abci/tests/test_behaviours.py:306]: When no longer specifying behaviour ids we need to use the class method to retrieve the auto generated id

File: packages/valory/skills/abstract_round_abci/base.py [hunk 4]
```diff
@@ -906,6 +908,26 @@ def _check_class_attributes(self) -> None:
...
+
+        This method returns the auto generated id from the class name if the
+        class variable behaviour_id is not set on the child class.
+        Otherwise, it returns the class variable behaviour_id.
+        """
+        return (
+            cls.round_id
+            if isinstance(cls.round_id, str)
+            else cls.__pattern.sub("_", cls.__name__).lower()
+        )
```
Comment (0xArdi) [inline packages/valory/skills/abstract_round_abci/base.py:924]: Same concern about conflicts as on the behaviour side. Does it make sense to include the module name?

Comment (DavidMinarsch) [reply packages/valory/skills/abstract_round_abci/base.py:924]: Same response

**Review by Adamantios — APPROVED** [review]
Similar to the behaviour refactoring, therefore I would like to comment on [the same](https://github.com/valory-xyz/open-autonomy/pull/1640#pullrequestreview-1213320197), but for `auto_round_id`.

~~`test_must_set_round_id` needs fixing.~~ I see it's been fixed on #1643.

**Pull request merged**